components/resource_metering/src/reporter.rs

```diff
@@ -159,7 +156,16 @@ impl Runnable for ResourceMeteringReporter {
                         |a, b| b.cmp(a),
                     );
                     let kth = self.find_top_k[self.config.max_resource_groups];
-                    self.records.retain(|_, (_, _, total)| *total >= kth);
+                    let others = &mut self.others;
+                    self.records.retain(|_, (secs_list, cpu_time_list, total)| {
+                        let retain = *total >= kth;
+                        if !retain {
+                            for (secs, cpu_time) in secs_list.iter().zip(cpu_time_list.iter()) {
+                                *others.entry(*secs).or_insert(0) += *cpu_time;
+                            }
+                        }
+                        retain
```

Comment:
How about using [`drain_filter`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain_filter), which seems to be, more functional?

Reply:
How someone come up with such a marvellous API...